populus/contracts/backends/filesystem.py

```diff
@@ -90,7 +90,12 @@ def get_contract_addresses(self, instance_identifier):
     #
     @property
     def registrar_path(self):
-        return self.config.get('file_path', './registrar.json')
+        file_path = self.config.get('file_path', './registrar.json')
+        project_dir = self.chain.project.project_dir
+        if os.path.dirname(os.path.abspath(file_path)) == project_dir:
```

Comment:
I think that this logic will fail if `file_path` is **both** an absolute path *and* nested in a subdirectory under the project dir.  The following should account for that case better.

```python
if os.path.isabs(file_path):
    abs_file_path = file_path
else:
    abs_file_path = os.path.abspath(os.path.join(project_dir, file_path))

if not is_under_path(project_dir, abs_file_path):
    raise ValueError("Registrar file path cannot be outside of populus project directory")
return abs_file_path
```

This allows for:

* declared absolute file paths
* file paths relative to the project directory

and it enforces that the file be under the project root which is good security imho.